More explicitly state that Windows 10 compatiblity will not be actively maintained (#3289)
wpilibsuite/frc-docs: source/docs/yearly-overview/yearly-changelog.rst

In order to more closly track C++ compiler feature support, the supported Linux distribution has changed to only the latest Ubuntu LTS. See :ref:`Supported Operating Systems and Architectures <docs/software/what-is-wpilib:Platform Support>` for more information.

> .. warning:: [Windows 10 support from Microsoft ended in October 2025](https://www.microsoft.com/en-us/windows/end-of-support). While we will not explicitly block Windows 10 from being used, future dependency updates may prevent you from using `wpilib` on Windows 10. We will not postpone these dependency updates in order to maintain Windows 10 compatibility, and Windows 10 support may break at any time as a result.

Reviewer: ```suggestion
.. warning:: [Windows 10 support from Microsoft ended in October 2025](https://www.microsoft.com/en-us/windows/end-of-support). While we will not explicitly block Windows 10 from being used, future changes may inadvertently break compatibility with Windows 10. We will not postpone these changes in order to maintain Windows 10 compatibility, and Windows 10 support may break at any time as a result.
```

Changed the wording a bit to broaden the scope to changes in general, not just dependency updates.

Reviewer 2: I think saying that it could break at any time is overly alarmist. It's not going to stop working without installing a new version. And as we typically don't do dependency updates or big changes mid-season, a mid-season update is also unlikely to break it. With the longer support timeline, an update between the FRC and FTC season is a little more likely, but by far the most likely time for it to break would be between major versions.

Reviewer 2: I like the updated language
